prometheus-cpp doesn't build

Hi,
I'm trying to build 0.4 and get this error:

```
[ 43%] Building CXX object lib/CMakeFiles/prometheus-cpp.dir/summary.cc.o
In file included from /hab/cache/src/prometheus-cpp-0.4/lib/summary.cc:1:0:
/hab/cache/src/prometheus-cpp-0.4/lib/../include/prometheus/summary.h:49:61: error: field 'quantiles_' has incomplete type 'const std::reference_wrapper<const std::vector<prometheus::detail::CKMSQuantiles::Quantile> >'
   const std::reference_wrapper<const std::vector<Quantile>> quantiles_;
                                                             ^~~~~~~~~~
In file included from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/bits/move.h:54:0,
                 from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/bits/stl_pair.h:59,
                 from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/utility:70,
                 from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/array:38,
                 from /hab/cache/src/prometheus-cpp-0.4/lib/../include/prometheus/summary.h:3,
                 from /hab/cache/src/prometheus-cpp-0.4/lib/summary.cc:1:
/hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/type_traits:2125:11: note: declaration of 'class std::reference_wrapper<const std::vector<prometheus::detail::CKMSQuantiles::Quantile> >'
     class reference_wrapper;
           ^~~~~~~~~~~~~~~~~
make[2]: *** [lib/CMakeFiles/prometheus-cpp.dir/build.make:303: lib/CMakeFiles/prometheus-cpp.dir/summary.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:155: lib/CMakeFiles/prometheus-cpp.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

This is using GCC 7.3.0, cmake 3.10.2, make 4.2.1, glibc 2.27

--- a/include/prometheus/summary.h
+++ b/include/prometheus/summary.h
@@ -3,6 +3,7 @@
 #include <array>
 #include <atomic>
 #include <chrono>
+#include <functional>
 #include <list>
 #include <mutex>
 #include <vector>

